A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js

```diff
@@ -347,6 +347,44 @@ var A320_Neo_UpperECAM;
                 }
             }
         }
+        getAPWarning(time) {
+            const count = time * 1000;
+            if (this.getCachedSimVar("AUTOPILOT MASTER", "Bool") == false) {
+                const timer = setTimeout(() => {
```

Comment:
Don't use setTimeout. You always want to use _deltaTime as that's provided by the sim and can deal with variable FPS much better and more accurately.

Reply:
> There's e a few issues with this PR: You should not be using setTimeout, and the way you use IsActive to change SimVars is prone to future breakage.
> 
> I would suggest moving your SimVar and timing logic into a separate function in this file, similar to say `updateInhibitMessages`, called by `update`. You'll then need some kind of timer to track the time differences, or you just use the NXLogic nodes for that. You can then set a variable that the `isActive` functions check in a read-only fashion.

Thx for your feedback. Currently, this is problem that calling caution and warning. I’ll build another logic for that.